pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -133,9 +133,7 @@ public HelixBrokerStarter(PinotConfiguration brokerConf, String clusterName, Str
 
   private int inferPort() {
     return Optional.ofNullable(_brokerConf.getProperty(Helix.KEY_OF_BROKER_QUERY_PORT)).map(Integer::parseInt)
```

Comment:
We don't need to check the `Helix.KEY_OF_BROKER_QUERY_PORT` because it will always be the first in the listenerConfigs if exists?
```suggestion
    return _listenerConfigs.get(0).getPort();
```

Reply:
fixed. I canned the entire method